noxfile.py

```diff
@@ -0,0 +1,6 @@
+import nox
```

Comment:
Should we maybe just remove top-level noxfile? I'm currently thinking `cargo xtask` can be all the top-level CI jobs, so plan to remove the makefile eventually too.